packages/connect/src/index.ts

```diff
@@ -1,3 +1,101 @@
+/**
+ * The substrate-connect package makes it possible to connect to Substrate-compatible blockchains.
+ *
+ * Connecting to a chain is done in two steps:
+ *
+ * - Call {getConnectorClient} or {createPolkadotJsScClient}, which gives you a so-called *client*.
+ * - Call `addChain` or `addWellKnownChain` on this client.
+ *
+ * Use {createPolkadotJsScClient} if you want to bind the PolkadotJS library to substrate-connect.
+ * This client will provide chains that implement the `ProviderInterface` interface defined in the
+ * `@polkadot/api` NPM package. These chains can be passed to `ApiPromise.create`, and you can
+ * then use the PolkadotJS library as you normally would. See <https://polkadot.js.org/docs>.
+ *
+ * Use {getConnectorClient} if you are interested in a lower-level experience where you directly
+ * send JSON-RPC requests and receive responses.
+ * In the library's internals, {createPolkadotJsScClient} is implemented on top of
+ * {getConnectorClient}.
+ *
+ * If you use {createPolkadotJsScClient}, be aware that the PolkadotJS library and its API are
+ * fundamentally built around full nodes functionnalities, while substrate-connect is a light
+ * client. Some functionnalities of PolkadotJS will not work properly, and/or not as well as they
+ * could. This is fundamentally not fixable, and, while {createPolkadotJsScClient} is acceptable
+ * for demos and prototypes, it is not possible to achieve the best user experience by using
+ * PolkadotJS. Proper light-client-oriented high-level libraries built on top of
+ * {getConnectorClient} are currently in development.
+ *
+ * # Adding parachains
+ *
+ * Connecting to a parachain is done the same way as connecting to a standalone chain: obtaining
+ * a client (with {getConnectorClient} or {createPolkadotJsScClient}) then calling `addChain`.
+ *
+ * However, if you call `addChain` with a parachain chain specification, you **must** have
+ * connected to its corresponding relay chain beforehand (using `addChain` or `addWellKnownChain`).
+ * Failing to do so will lead to an error at the initialization of the parachain.
+ *
+ * Furthermore, the parachain must be added to the same client object as the one the relay chain
+ * was added to.
+ *
+ * In other words, this will work:
+ *
+ * ```js
+ * const client = getConnectorClient();
+ * await client.addChain(relayChain);
```

Comment:
Should there be mentioned in all these examples the `JSON.stringify(....)` or its irrelevant for current example-show-case?

Reply:
The point of this example is to illustrate what the text says, not to demonstrate how to use the library